web/src/services/api.ts

```diff
@@ -343,6 +334,27 @@ export abstract class Api {
             });
         };
     }
+    private static getClientIP(req: Request) {
+        var client_ip = (typeof req.headers['x-forwarded-for'] === 'string'
+                && req.headers['x-forwarded-for'].split(',').shift())
+                || req.connection?.remoteAddress;
+        if(client_ip && client_ip.includes("::ffff:")){
+            client_ip = client_ip.replace("::ffff:", "")
+        }
+        return client_ip;
+    }
+    private static setHeaders(req: Request, headers: any) {
+        const requestData = req && req.body ? JSON.stringify(req.body) : "";
+        var client_ip = Api.getClientIP(req);
+        let authorization = req.headers ? (req.headers.authorization ? req.headers.authorization : "") : "";
```

Comment:
No need to check req.headers every time. One time should be enough

Reply:
Done